dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/datastore/FileDataStoreHandler.kt

```diff
@@ -0,0 +1,289 @@
+import com.datadog.android.core.internal.persistence.datastore.ext.toByteArray
+import com.datadog.android.core.internal.persistence.datastore.ext.toInt
+import com.datadog.android.core.internal.persistence.datastore.ext.toLong
+import com.datadog.android.core.internal.persistence.file.FileReaderWriter
+import com.datadog.android.core.internal.persistence.file.createNewFileSafe
+import com.datadog.android.core.internal.persistence.file.deleteSafe
+import com.datadog.android.core.internal.persistence.file.existsSafe
+import com.datadog.android.core.internal.persistence.file.mkdirsSafe
+import com.datadog.android.core.internal.persistence.tlvformat.FileTLVBlockReader
+import com.datadog.android.core.internal.persistence.tlvformat.TLVBlock
+import com.datadog.android.core.internal.persistence.tlvformat.TLVBlockType
+import com.datadog.android.core.persistence.Serializer
+import java.io.File
+import java.util.Locale
+
+@Suppress("TooManyFunctions")
+internal class FileDataStoreHandler(
+    private val sdkInstanceId: String,
+    private val storageDir: File,
+    private val internalLogger: InternalLogger,
+    private val fileReaderWriter: FileReaderWriter,
+    private val fileTLVBlockReader: FileTLVBlockReader,
+    private val dataStoreFileHelper: DataStoreFileHelper
+) : DataStoreHandler {
+    @WorkerThread
+    override fun <T : Any> write(
+        dataStoreFileName: String,
+        featureName: String,
+        serializer: Serializer<T>,
+        data: T
+    ) {
+        val dataStoreDirectory = createDataStoreDirectoryIfNecessary(featureName)
+        val dataStoreFile = createDataStoreFileIfNecessary(dataStoreDirectory, dataStoreFileName)
+
+        val lastUpdateBlock = getLastUpdateDateBlock()
+        val versionCodeBlock = getVersionCodeBlock()
+        val dataBlock = getDataBlock(data, serializer)
+
+        if (lastUpdateBlock == null || versionCodeBlock == null || dataBlock == null) return
+
+        writeToFile(
+            dataStoreFile,
+            lastUpdateBlock + versionCodeBlock + dataBlock
+        )
+    }
+
+    @WorkerThread
+    override fun <T : Any> read(
+        dataStoreFileName: String,
+        featureName: String,
+        deserializer: Deserializer<String, T>,
+        version: Int
+    ): T? {
+        val dataStoreDirectory = dataStoreFileHelper.getDataStoreDirectory(
+            sdkInstanceId = sdkInstanceId,
+            featureName = featureName,
+            folderName = DATASTORE_FOLDER_NAME.format(Locale.US, version),
+            storageDir = storageDir
+        )
+
+        val datastoreFile = dataStoreFileHelper.getDataStoreFile(
+            dataStoreDirectory = dataStoreDirectory,
+            dataStoreFileName = dataStoreFileName
+        )
+
+        if (!datastoreFile.existsSafe(internalLogger)) {
+            return null // no datastore file found
+        }
+
+        return readFromDataStoreFile(datastoreFile, deserializer, fileTLVBlockReader, version)
+    }
+
+    @WorkerThread
```

Comment:
do we need annotation on all these private methods? all the public entry points in this class already have `@WorkerThread` annotations.